doc/introduction/compiling_circuits.rst

```diff
@@ -332,3 +339,69 @@ observables and coefficients:
 [[0.97, 4.21], [1.43]]
 
 This and more logic to manipulate Pauli observables is found in the :mod:`~.pennylane.grouping` module.
+
+
+Just-in-time compilation with Catlyst
+-------------------------------------
+
+In addition to quantum circuit transformations, PennyLane also supports full
+hybrid just-in-time (JIT) compilation via `Catalyst
+<https://github.com/pennylaneai/catalyst>`__. Catalyst allows you to compile
+the entire quantum-classical workflow, including any optimization loops,
+which allows for optimized performance, and the ability to run the entire
+workflow on accelerator devices as appropriate.
+
+Currently, Catalyst must be installed separately, and only supports the JAX
+interface and ``lightning.qubit``. Check out the Catalyst documentation for
+`installation instructions
+<https://docs.pennylane.ai/projects/catalyst/en/latest/dev/installation.html>`__.
+
+Using Catalyst with PennyLane is a simple as using the ``@qjit`` decorator to
+compile your hybrid workflows:
+
+.. code-block:: python
+
+    from catalyst import qjit
+    from jax import numpy as jnp
+
+    dev = qml.device("lightning.qubit", wires=2, shots=1000)
+
+    @qml.qnode(dev)
+    def cost(params):
+        qml.Hadamard(0)
+        qml.RX(jnp.sin(params[0]) ** 2, wires=1)
+        qml.CRY(params[0], wires=[0, 1])
+        qml.RX(jnp.sqrt(params[1]), wires=1)
+        return qml.expval(qml.PauliZ(1))
+
+The ``qjit`` decorator can also be used on hybrid cost functions -- that is,
+cost functions that include both QNodes and classical processing. We can even
+JIT compile the full optimization loop, for example when training models:
+
+.. code-block:: python
+
+    import jaxopt
+
+    @jax.jit
+    def optimization():
+        # initial parameter
+        params = jnp.array([0.54, 0.3154])
```

Comment:
I'm a bit torn here with this example because it's talking about the qjit decorator compiling hybrid cost functions but then we use JAX to do it in the example, which is supported with the new JAX integration and convenient, but still a bit strange 🤔

Reply:
Yep I'm torn as well. The one advantage is that you avoid having to create a wrapper around the cost function which returns cost _and_ gradient, and then making use of the equivalent jaxopt feature 🤔

Reply:
Sounds good, we'll leave it like this for now then.